packages/indexer-agent/src/commands/start.ts

```diff
@@ -37,6 +37,9 @@ import { displayZodParsingError } from './error-handling'
 // eslint-disable-next-line  @typescript-eslint/no-explicit-any
 export type AgentOptions = { [key: string]: any } & Argv['argv']
 
+const DEFAULT_SUBGRAPH_MAX_BLOCK_DISTANCE = 0
```

Comment:
0 block distance sounds good, just wondering if it could be loosened to 2 or 5 or so for the default behavior? for subgraphs deployed locally, what about using `synced` from the indexing statuses query instead of calculating the block number freshness?

Reply:
Based on our manual tests, Ethereum protocol subgraphs were synced most of the time (block distance = 0), so we didn't set the default to a higher value. 

I think we could use the `synced` attribute, but we would need to add an extra call to Graph Node just to check the synced status. With the current design, we keep the same number of calls (graphql queries, in this case) as before.

I'm open to discussing the benefits of either approach.

Reply:
Makes sense, thank you, this is more of a curiosity question than anything

To me, I can see that
Graph node status approach 
- Pros: network agnostic, no need for block calculation and so no need for max_block_distance config, graph node and indexer would be on the same page with `sync` status abstracting away the rpc providers
- Cons: additional call to graph node, may affect query service performance

Current design
- Pros: keep the same number of graphql queries made to Graph node, control of freshness requirement falls to indexer instead of graph node status calculation
- Cons: may require more granular control over max_block_instance configuration for specific networks, require indexers to understand freshness requirement 

I think either way is good and can be discussed in a later issue